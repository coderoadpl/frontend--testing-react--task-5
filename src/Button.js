import React from 'react'
import PropTypes from 'prop-types'

export const Button = (props) => {
  const {
    children,
    ...otherProps
  } = props

  return (
    <button
      style={{ borderRadius: '50%' }}
      {...otherProps}
    >
      {children}
    </button>
  )
}

Button.propTypes = {
  children: PropTypes.node
}

export default Button
