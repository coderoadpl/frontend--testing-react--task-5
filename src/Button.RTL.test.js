import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import Button from './Button'

describe('Button test â€“ React Testing Library', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Button />)
    expect(asFragment()).toMatchSnapshot()
  })
})
