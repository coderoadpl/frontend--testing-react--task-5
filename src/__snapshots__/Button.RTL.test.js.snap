// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test â€“ React Testing Library should match snapshot 1`] = `
<DocumentFragment>
  <button
    style="border-radius: 50%;"
  />
</DocumentFragment>
`;
